### PR TITLE
Issue #3170: delete children from javadoc TEXT

### PR DESCRIPTION
**Pull request by rnveach**
Issue #3170

Main modification is in `JavadocNodeImpl.java`. All other changes are reaction to that change.
Change to `EMPTY_DETAIL_NODE_ARRAY` is needed to avoid a class cast exception.

Modification in `TokenTypes.java` was found in a kind of bug in `JavadocParagraphCheck.java`. This is why it now checks for `TEXT` token.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocNodeImpl.java [hunk 1]
```diff
@@ -94,7 +95,9 @@ public int getColumnNumber() {
 
     @Override
     public DetailNode[] getChildren() {
-        if (children == null) {
+        if (children == null
+                // remove excess tokens
+                || type == JavadocTokenTypes.TEXT) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocNodeImpl.java]: I do not like such hack. 
We need to change tree right after parse. JavadocNodeImpl should stay simple POJO data container.

Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocNodeImpl.java]: This was moved to the class that called it, `JavadocDetailNodeParser`.

**Comment by rnveach** [comment]
Regression: http://rveach.no-ip.org/checkstyle/regression/reports/108/

`RxJava` can be ignored. It is a known issue with javadocs.
Only true difference is at: http://rveach.no-ip.org/checkstyle/regression/reports/108/checkstyle/index.html#A1

**Comment by romani** [comment]
Regression in checkstyle code is ok.

Rxjava regression is happend due to different order of Checks during execution. Message is the same.

**Comment by romani** [comment]
@rnveach ,

1)
there are still items to fix.
```
$ grep -Rl "\-\-CHAR" src/
src/test/resources/com/puppycrawl/tools/checkstyle/grammars/InputRegressionJavaClass1Ast.txt
src/xdocs/writingjavadocchecks.xml.vm

```

2)
redo checkstyle/src/site/resources/images/gui_javadoc_screenshot.png

**Comment by rnveach** [comment]
@romani 
> grammars/InputRegressionJavaClass1Ast

This looks like a false positive since this is java only regression. The only `CHAR` I see in there are `CHAR_LITERAL` and `LITERAL_CHAR`. We only have 1 javadoc in that file and it isn't in the AST output.

The others I can confirm and will fix.

**Comment by romani** [comment]
right, updated command base on master code 
```
$ grep -Rl "\-\-CHAR ->" src/
src/test/resources/com/puppycrawl/tools/checkstyle/astprinter/expectedInputJavadocComment.txt
src/test/resources/com/puppycrawl/tools/checkstyle/astprinter/expectedInputAstTreeStringPrinterJavaAndJavadoc.txt
src/test/resources/com/puppycrawl/tools/checkstyle/astprinter/expectedInputAstTreeStringPrinterJavadoc.txt
src/test/resources/com/puppycrawl/tools/checkstyle/astprinter/expectedInputAstTreeStringPrinterAttributesAndMethodsJavadoc.txt
src/xdocs/writingjavadocchecks.xml.vm
```

**Comment by rnveach** [comment]
@romani Both points were done.
The only file that still had `-- CHAR ->` left was `writingjavadocchecks`.

**Comment by rnveach** [comment]
Change to `CodeSelectorPModelTest` was because the specified test picked a `CHAR` to use and it had been removed. Split the line into 2 variables so it is easier to see where the fail is and what it is looking for.

**Pull request merged**